### DISCUSSION TITLE
[v4.1] How to stop Redis from preventing my script to end?

### DISCUSSION THREAD

**Discussion by mathieuh** [adonisjs/core#1584] [Help] [answered]
Hi,

In one of my adonis command I use Redis and this command never ends.
I know it's Redis because when I comment it in my command it ends properly.
I tried to call `Redis.quit()` at the end of the command but it still doesn't end.

Is it a bug? Any idea what I'm doing wrong?

Cheers,

**Comment by thetutlage** [comment]
Can you share the code for your command?

**Reply by mathieuh** [reply]
Here it is:
```
'use strict'

const { Command } = require('@adonisjs/ace')
const Caterer = use('App/Models/Caterer')
const Order = use('App/Models/Order')
const Redis = use('Redis')
const Database = use('Database')

class UpdateFooterNumber extends Command {
  static get signature () {
    return 'update:footer:number'
  }

  static get description () {
    return 'Met à jour les chiffres du footer'
  }

  async handle (args, options) {
    const count_orders = await Order.query().where('status', 'accepted').getCount()
    // affiner en faisant une moyenne du nombre de participant par prestation
    const footer_numbers = {
        meal: count_orders * 10,
        tree: count_orders,
        caterer: await Caterer.query().where('active', 1).getCount()
    }
    await Redis.set('footer_numbers', JSON.stringify(footer_numbers))
    Database.close()
    this.info('Les chiffres du footer ont été mis à jour')
  }
}

module.exports = UpdateFooterNumber
```

**Reply by thetutlage** [reply]
I don't see any code closing the Redis connection? You should do the following

```ts
await Database.close()
await Redis.quit()
```

**Reply by mathieuh** [reply]
Yeah that's what I did, I removed it from my example because it didn't seem to work.
I'll try to make a repo that reproduce the issue and link it here.

**Reply by thetutlage** [reply]
Sure!

**Reply by mathieuh** [reply]
https://github.com/mathieuh/adonis-core-1584
Finally, sorry for the delay.
You can reproduce with `adonis update:user`.
Uses mysql and redis.

**Reply by thetutlage** [reply]
Yup, seems like some issue with the `Redis.quit` method. For now you can use `Redis.disconnect` and I tested that it works. The disconnect method forcefully closes the connection, whereas quit performs a graceful disconnect.

```ts
await Database.close()
await Redis.disconnect()
```

**Reply by mathieuh** [reply]
I didn't know about `Redis.disconnect()`.
Don't know what `Redis.quit()` is waiting for though, I'll try to investigate.

Closing this, Thanks.

**Comment by mathieuh** [comment, marked as answer]
Like the doc says, Redis.quit() is actually waiting for the connection name.
https://adonisjs.com/docs/4.1/redis#_quitname

Redis.quit('local') if working locally for example.

My bad.
